hotel/src/hil/rng.rs

```diff
@@ -0,0 +1,25 @@
+/// A Random Number Generator, based on the Rng trait from the rand crate.
+pub trait Rng {
+    /// Returns 4 bytes of random data.  Types that implement the Rng trait
+    /// only need to implement this function.
+    fn next_u32(&mut self) -> u32;
```

Comment:
I think your suggestion is quite elegant.  However, the part I'm struggling with is: if an application provides a large buffer, how do I block it from scheduling until the buffer has been filled without an allow / subscribe / yield from the application?